Resolves: #1123093

--- a/scripts/logwatch.pl	2016-03-16 14:27:25.770098250 +0100
+++ b/scripts/logwatch.pl	2016-03-16 14:28:46.689067743 +0100
@@ -1469,14 +1469,15 @@ sub parselogs {
    if (defined fileno OUTFILE) {
       print OUTFILE $out_mime if $out_mime;
       if ( $Config{'encode'} eq "base64" ) {
-         print OUTFILE encode_base64($out_head) if $out_head;
-         print OUTFILE encode_base64($out_reference) if $out_reference;
+         my $out = '';
+         $out .= $out_head if $out_head;
+         $out .= $out_reference if $out_reference;
          foreach ( 0 .. $index_par ) {
-            print OUTFILE encode_base64($out_body{$_}) if defined( $out_body{$_} );
-#fixme
+            $out .= $out_body{$_} if defined( $out_body{$_} );
             $out_body{$_} = ''; #We should track this down out_body could be an array instead also -mgt
          }
-         print OUTFILE encode_base64($out_foot) if $out_foot;
+         $out .= $out_foot if $out_foot;
+         print OUTFILE encode_base64($out);
       } else {
          print OUTFILE $out_head if $out_head;
          print OUTFILE $out_reference if $out_reference;
