diff -rup logwatch-7.3.6/scripts/services/sshd logwatch-7.3.6-new/scripts/services/sshd
--- logwatch-7.3.6/scripts/services/sshd	2016-09-13 13:41:13.195660428 +0200
+++ logwatch-7.3.6-new/scripts/services/sshd	2016-09-13 13:43:00.706687702 +0200
@@ -576,7 +576,7 @@ if (keys %KrbErr) {
    }
 }
 
-if ( ($Detail == 7 && keys %KrbRealm > 1) || ($Detail > 8 && keys %KrbRealm) ){
+if ( ($Detail == 7 && keys %KrbRealm > 1) || ($Detail >= 8 && keys %KrbRealm) ){
    print "\nSucessfull Kerberos Authentication from ",(scalar keys %KrbRealm)," realm:\n";
    foreach my $realm (keys %KrbRealm) {
       if($Detail > 9){
