From e16d1363b33344c53da355983445505843c3cfe1 Mon Sep 17 00:00:00 2001
From: Jan Synacek <jsynacek@redhat.com>
Date: Thu, 17 Mar 2016 09:57:25 +0100
Subject: [PATCH 2/4] conf: fix disrupted sentence

---
 conf/logwatch.conf | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/conf/logwatch.conf b/conf/logwatch.conf
index 6faae8f..e031a6b 100644
--- a/conf/logwatch.conf
+++ b/conf/logwatch.conf
@@ -60,10 +60,9 @@ MailFrom = Logwatch
 # (i.e. /var/log/messages.1 or /var/log/messages.1.gz) will
 # be searched in addition to the /var/log/messages file.
 # This usually will not do much if your range is set to just
-# 'Yesterday' or 'Today'... it is probably best used with
+# 'Yesterday' or 'Today'... it is probably best used with Range = All.
 # By default this is now set to Yes. To turn off Archives uncomment this.
 #Archives = No
-# Range = All
 
 # The default time range for the report...
 # The current choices are All, Today, Yesterday
-- 
2.7.3

