From 7841afe66d5b2049b229e8b8faddafc84aa08b68 Mon Sep 17 00:00:00 2001
From: Jan Synacek <jsynacek@redhat.com>
Date: Thu, 17 Mar 2016 09:58:46 +0100
Subject: [PATCH 3/4] puppet: improve counting of successful runs

---
 scripts/services/puppet | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/services/puppet b/scripts/services/puppet
index 305dfff..1b86ae7 100644
--- a/scripts/services/puppet
+++ b/scripts/services/puppet
@@ -66,6 +66,8 @@ while (defined($ThisLine = <STDIN>)) {
       }
    } elsif ($ThisLine =~ /Finished catalog run in [0-9]+\.[0-9]+ seconds/) {
       $SuccessfulRuns++;
+   } elsif ($ThisLine =~ /Applied catalog in [0-9]+\.[0-9]+ seconds/) {
+      $SuccessfulRuns++;
    } elsif ($ThisLine =~ /skipping run/) {
       $FailedRuns++;
    } elsif ($ThisLine =~ /(Did not receive certificate)/) {
-- 
2.7.3

